File: cari-backend/src/models/User.model.js
const bcrypt = require('bcryptjs');
const mongoose = require('mongoose');

const SALT_ROUNDS = 10;

const userSchema = new mongoose.Schema(
  {
    email: {
      type: String,
      required: true,
      unique: true,
      lowercase: true,
    },
    password: {
      type: String,
      required: true,
    },
    companyName: { type: String, default: '' },
    taxOffice: { type: String, default: '' },
    taxId: { type: String, default: '' },
    companyPhone: { type: String, default: '' },
    companyAddress: { type: String, default: '' },
  },
  { timestamps: true }
);

userSchema.pre('save', async function hashPassword() {
  if (!this.isModified('password')) {
    return;
  }
  this.password = await bcrypt.hash(this.password, SALT_ROUNDS);
});

/**
 * @param {string} enteredPassword
 * @returns {Promise<boolean>}
 */
userSchema.methods.matchPassword = async function matchPassword(enteredPassword) {
  return bcrypt.compare(enteredPassword, this.password);
};

module.exports = mongoose.model('User', userSchema);

File: cari_flutter/lib/features/auth/data/models/user_model.dart
class UserModel {
  const UserModel({
    required this.id,
    required this.email,
    this.companyName = '',
    this.taxOffice = '',
    this.taxId = '',
    this.companyPhone = '',
    this.companyAddress = '',
  });

  final String id;
  final String email;
  final String companyName;
  final String taxOffice;
  final String taxId;
  final String companyPhone;
  final String companyAddress;

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: (json['id'] ?? json['_id'] ?? '').toString(),
      email: (json['email'] ?? '').toString(),
      companyName: (json['companyName'] ?? '').toString(),
      taxOffice: (json['taxOffice'] ?? '').toString(),
      taxId: (json['taxId'] ?? '').toString(),
      companyPhone: (json['companyPhone'] ?? '').toString(),
      companyAddress: (json['companyAddress'] ?? '').toString(),
    );
  }

  UserModel copyWith({
    String? id,
    String? email,
    String? companyName,
    String? taxOffice,
    String? taxId,
    String? companyPhone,
    String? companyAddress,
  }) {
    return UserModel(
      id: id ?? this.id,
      email: email ?? this.email,
      companyName: companyName ?? this.companyName,
      taxOffice: taxOffice ?? this.taxOffice,
      taxId: taxId ?? this.taxId,
      companyPhone: companyPhone ?? this.companyPhone,
      companyAddress: companyAddress ?? this.companyAddress,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'email': email,
      'companyName': companyName,
      'taxOffice': taxOffice,
      'taxId': taxId,
      'companyPhone': companyPhone,
      'companyAddress': companyAddress,
    };
  }
}
